### DISCUSSION TITLE
Child node types restriction settings?

### DISCUSSION THREAD

**Discussion by polymed** [pimcore/pimcore#8640] [Q&A / Support] [answered]
Hi guys,

Is it possible to limit which child node types a node can have?

For example: `Class A` can only have Child Nodes of Type `Class A`

Best regards

**Comment by NiklasBr** [comment, marked as answer]
It's not possible in the UI out of the box, but with [Event Listeners](https://pimcore.com/docs/pimcore/current/Development_Documentation/Extending_Pimcore/Event_API_and_Event_Manager.html) on Data Objects (my guess is `PRE_ADD` and `PRE_UPDATE`) you should be able to inspect the object with `getParent()` and handle it from there to allow or not.

**Reply by polymed** [reply]
I thought so, thank you very much
